rlp/split_test.go

```diff
@@ -0,0 +1,91 @@
+package rlp
+
+import (
+	"bytes"
+	"reflect"
+	"testing"
+)
+
+func TestCountValues(t *testing.T) {
+	tests := []struct {
+		input string
+		count int
+		err   error
+	}{
+		{"", 0, nil},
+		{"00", 1, nil},
+		{"80", 1, nil},
+		{"C0", 1, nil},
+		{"010203", 3, nil},
+		{"01C40607080902", 3, nil},
+		{
+			input: "A12000BF49F440A1CD0527E4D06E2765654C0F56452257516D793A9B8D604DCFDF2AB853F851808D10000000000000000000000000A056E81F171BCC55A6FF8345E692C0F86E5B48E01B996CADC001622FB5E363B421A0C5D2460186F7233C927E7DB2DCC703C0E500B653CA82273B7BFAD8045D85A470",
+			count: 2,
+			err:   nil,
+		},
+	}
```

Comment:
Also, I think testing both sides of the boundary values would be useful (i.e. 0x7f, 0x80, 0xb7, 0xb8, 0xbf, 0xc0, 0xf7, 0xf8, 0xff), though there might be valid issues with the big lengths, so maybe only do 0x7f, 0xb8, 0xf8.